File: Python/Easy/37.py
# Daily Coding Problem: 37
# Difficulty: Easy
# Asked by: Google
# Author: TenthEdict

def power_set(arr):
    n = len(arr)
    result = []

    for i in range(1 << n):
        subset = []

        for j in range(n):
            if i & (1 << j) != 0 : 
                subset.append(arr[j])
       
        result.append(subset)
    
    return result

print(power_set([1, 2, 3]))
print(power_set([1, 2]))
# print(power_set([]))

File: JavaScript/Easy/37.js
// Daily Coding Problem: 37
// Difficulty: Easy
// Asked By: Google
// Author: TenthEdict

function power_set(arr) {
    let n = arr.length;
    let result = [];

    for (let i = 0; i < (1 << n); i++) {
        let subset = [];

        for (let j = 0; j < n; j++) {
            if ((i & (1 << j)) != 0) {
                subset.push(arr[j]);
            }
        }
        result.push(subset);
    }
    return result;
}